src/network/networkd-json.c

```diff
@@ -1170,6 +1171,31 @@ static int dhcp_client_pd_append_json(Link *link, JsonVariant **v) {
         return json_variant_set_field_non_null(v, "6rdPrefix", array);
 }
 
+static int dhcp_client_private_options_append_json(Link *link, JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp_lease)
+                return 0;
+
+        LIST_FOREACH(options, option, link->dhcp_lease->private_options) {
+                char key[STRLEN("Option_000")+1];
+
+                xsprintf(key, "OPTION_%" PRIu8, option->tag);
```

Comment:
why turn this into a string? this is a numer value, hence expose as that?

Reply:
I am trying to construct the private options in the same way how it is being stored in the lease file. So when we query dbus it would like below.

PrivateOptions\":[{\"OPTION_246\":\"XXXX\"},{\"OPTION_247\":\"XXXX\"}]

Reply:
This is structured data so it doesn't need to be in string format, it's already part of an array called "options" so it's redundant to say "option_xxx". Data should be exposed in its natural type format

Reply:
I think it is better to store the array of code and data, like `dhcp6_client_vendor_options_append_json()` does.

Reply:
@yuwata : If I am correct you wanted to have a getter function. correct me if I am wrong.

Reply:
Yeah, I'd like to also request that. But here requested thing is the output format.

Reply:
I fixed it like how it was done for dhcp6_client_vendor_options_append_json(). It would look like below now in dbus.

\"PrivateOptions\":[{\"Option\":246,\"PrivateOptionData\":\"XXXX\"},{\"Option\":247,\"PrivateOptionData\":\"XXXX\"}

Kindly let me know if this is acceptable.